ecosystem/sep-0030.md

```diff
@@ -76,6 +75,52 @@ A client can also use the endpoints in the specification to:
 - Update the identities associated with an account.
 - Delete accounts.
 
+## Use Cases
+
+This protocol is intended to satisfy a wide range of use cases, but initially
+the two use cases mentioned under [Motivation] are apparent.
+
+### Use Case: Recover
+An individual that expects to maybe lose access to their keys can use SEP-30 to
+recover access to their Stellar accounts.
+
+A single identity is sufficient and specifying a role would be unnecessary. The
+individual registers the account with two or more servers and gives each server
+signer weights so that no individual server can control the account, but
+together they can approve transactions.
+
+The individual can recover the account by authenticating with all servers and
+requesting all servers sign a transaction giving a new key signing weight
+sufficient to meet the high threshold of the account.
+
+### Use Case: Share
+An individual that wishes to share an account with another user and allow that
+other user to gain access to the account by recovering it from two or more
+servers.
+
+Multiple identities should be specified each with a unique role so that when
+each individual is authenticated using an identities authentication methods
+they can recognize their role in relation to the account. In this case using
+roles such as `sender` and `receiver` are appropriate.
+
+Either individual can recover the account by authenticating with all servers and
+asking all servers to sign a transaction giving a new key signing weight
+sufficient to meet the high threshold of the account.
+
+The receiver can accept the account and either take ownership of it or merge it
```

Comment:
The current [sign](https://github.com/leighmcculloch/stellar--stellar-protocol/blob/6628c066d085585ac93ca179a03df13602bf179d/ecosystem/sep-0030.md#post-accountsaddresssign) endpoint is not expecting any operation other than `SET_OPTIONS`. Should we update it?

Reply:
>The transaction can contain any operations

I think I'm missing something about the request. The sign endpoint can accept any operation. What at this line should we update?

Reply:
I'm not sure how we plan on implementing this endpoint. According to

> The transaction must only contain operations for the account. The transaction can contain any operations, but it is anticipated for the use cases discussed earlier in this protocol that the transaction will contain an operation to add a new signer to the account and possibly to remove an old signer.

It is not clear to me what it means by `contain operations for the account`? Does receiving a payment from other account or sending payment to another account consider operations for only the account?

On top of that, given that this service is currently called recoverysigner, I think it might be more explicit that it only does adding and removing signers of a specific account. And we will let the account decide what to do with the signer power.

Reply:
> It is not clear to me what it means by contain operations for the account? Does receiving a payment from other account or sending payment to another account consider operations for only the account?

Nvm. I found the answer in the code: https://github.com/stellar/go/pull/2383/files#diff-28eb4357c2c210cfb8f53d9dbc938aa3R112-R127

We might just want to make it clear what we mean by `only contain operations for the account` by saying `the source account of the transaction and the source account of all operations in the transaction must match the account in the request.`

Reply:
=> #572